### ISSUE TITLE
F-Droid can't build

### ISSUE THREAD

**Issue by licaon-kter** [openvk/mobile-android-legacy#58]
9e6f9f754571895a12743deb261bcd106cad50cc

```
...
:app:compileReleaseSources
:app:lintVitalRelease/home/vagrant/build/uk.openvk.android.legacy/app/src/main/res/layout/tab_selector.xml:55: Error: Duplicate id @+id/tab_text, already defined earlier in this layout [DuplicateIds]
                android:id="@+id/tab_text"
                ~~~~~~~~~~~~~~~~~~~~~~~~~~
    /home/vagrant/build/uk.openvk.android.legacy/app/src/main/res/layout/tab_selector.xml:24: Duplicate id @+id/tab_text originally defined here
/home/vagrant/build/uk.openvk.android.legacy/app/src/main/res/layout/tab_selector.xml:70: Error: Duplicate id @+id/tab_checked, already defined earlier in this layout [DuplicateIds]
                android:id="@+id/tab_checked"
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /home/vagrant/build/uk.openvk.android.legacy/app/src/main/res/layout/tab_selector.xml:39: Duplicate id @+id/tab_checked originally defined here
/home/vagrant/build/uk.openvk.android.legacy/app/src/main/res/layout/tab_selector.xml:87: Error: Duplicate id @+id/tab_text, already defined earlier in this layout [DuplicateIds]
                android:id="@+id/tab_text"
                ~~~~~~~~~~~~~~~~~~~~~~~~~~
    /home/vagrant/build/uk.openvk.android.legacy/app/src/main/res/layout/tab_selector.xml:24: Duplicate id @+id/tab_text originally defined here
/home/vagrant/build/uk.openvk.android.legacy/app/src/main/res/layout/tab_selector.xml:102: Error: Duplicate id @+id/tab_checked, already defined earlier in this layout [DuplicateIds]
                android:id="@+id/tab_checked"
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /home/vagrant/build/uk.openvk.android.legacy/app/src/main/res/layout/tab_selector.xml:39: Duplicate id @+id/tab_checked originally defined here

   Explanation for issues of type "DuplicateIds":
   Within a layout, id's should be unique since otherwise findViewById() can
   return an unexpected view.

4 errors, 0 warnings
 FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:lintVitalRelease'.
> Lint found fatal errors while assembling a release target.
  
  To proceed, either fix the issues identified by lint, or modify your build script as follows:
  ...
  android {
      lintOptions {
          checkReleaseBuilds false
          // Or, if you prefer, you can continue to check for errors in release builds,
          // but continue the build even when errors are found:
          abortOnError false
      }
  }
  ...
```

**Comment by tretdm** [comment]
fixed in [58a8f10](https://github.com/openvk/mobile-android-legacy/commit/58a8f1044d1b88f97df8d69323f11261913d29a2)

**Comment by licaon-kter** [comment]
Or https://github.com/openvk/mobile-android-legacy/commit/3c8ce99b85808470f5893a02b9c7591255631285 :)

**Comment by linsui** [comment]
Could you please also fix the MissingTranslation lint error?

**Comment by tretdm** [comment]
> Could you please also fix the MissingTranslation lint error?

later

**Comment by licaon-kter** [comment]
build 169: 
```
:app:mergeBetaConfigReleaseResources
/home/vagrant/build/uk.openvk.android.legacy/app/src/main/res/values-v7/style.xml: Error: Found item Style/BaseStyle.DialogTheme more than one time
:app:mergeBetaConfigReleaseResources FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeBetaConfigReleaseResources'.
> /home/vagrant/build/uk.openvk.android.legacy/app/src/main/res/values-v7/style.xml: Error: Found item Style/BaseStyle.DialogTheme more than one time
```

**Comment by tretdm** [comment]
build 170 soon, the fix will also be

**Comment by tretdm** [comment]
check: https://github.com/openvk/mobile-android-legacy/commit/195fbbf6959c88f1c65a7f8d5c72764e834f157a

**Comment by licaon-kter** [comment]
Soooo old?

**Comment by tretdm** [comment]
Android NDK download links (Linux x86_64):
[Android NDK r8e](http://dl.google.com/android/ndk/android-ndk-r8e-linux-x86_64.tar.bz2)
[Android NDK r11c](https://dl.google.com/android/repository/android-ndk-r11c-linux-x86_64.zip)

**Comment by tretdm** [comment]
@licaon-kter 
![изображение](https://github.com/openvk/mobile-android-legacy/assets/76806170/e849e1f4-57d3-445c-bd27-f130b82e3e22)

Well, that's the reason why you should use r11c when building your APK, as newer versions have increased the minimum Android API level threshold from 3 to 8.

https://stackoverflow.com/a/68856400

**Comment by licaon-kter** [comment]
@tretdm 

Where do you get [ndk-build.sh](https://github.com/openvk/mobile-android-legacy/blob/1.1.203-beta/app/build.gradle#L139) from? It's not in any NDK I tried. So I renamed it to `ndk-build` and works.

The relative path needs to be `$projDir/../ndk-modules/ovkmplayer` here: https://github.com/openvk/mobile-android-legacy/blob/1.1.203-beta/app/build.gradle#L142-L143 because `projDir` is `app` afaics

updated recipe: https://gitlab.com/fdroid/fdroiddata/-/commit/11d15cc3b902e1a6983fc3d5aedf50db9d0209a0

**Comment by tretdm** [comment]
> Where do you get [ndk-build.sh](https://github.com/openvk/mobile-android-legacy/blob/1.1.203-beta/app/build.gradle#L139) from?

@licaon-kter Line [129](https://github.com/openvk/mobile-android-legacy/blob/1.1.203-beta/app/build.gradle#L129) points to the path to the Android NDK specified in the project structure in Android Studio.

**Comment by tretdm** [comment]
Ok, corrected in commit https://github.com/openvk/mobile-android-legacy/commit/32f50cb9aaf7c06f3e56c1af18f5c2b1999ca37a

**Comment by tretdm** [comment]
So wait, in this commit, for some reason, the FFmpeg binaries are missing when building.

**Comment by tretdm** [comment]
Fixed in b722dbf. Note that the `app` directory is created inside `app` - it is necessary that the collected files are not erased during compilation, otherwise an error will occur.

**Comment by tretdm** [comment]
FFmpeg version for 32-bit CPU architectures (ARMv5/ARMv7/x86) upgraded from **0.8.12** to **0.11.5**.

```bash
chmod +x ./build-ffmpeg.sh && \
ANDROID_NDK_HOME=[path/to/ndk-r8e] ./build-ffmpeg.sh 0.11.5 && \
ANDROID_NDK_HOME=[path/to/ndk-r11c] ./build-ffmpeg.sh 3.1.4
```
